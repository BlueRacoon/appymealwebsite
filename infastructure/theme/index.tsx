// infastructure/theme/index.ts
import { colors } from "./colors";
import { space, lineHeights } from "./spacing";
import { sizes } from "./sizes";
import { fonts, fontWeights, fontSizes } from "./fonts";
import { Theme } from "./theme.types";

export const theme: Theme = {
  colors,
  space,
  lineHeights,
  sizes,
  fonts,
  fontSizes,
  fontWeights,
};
