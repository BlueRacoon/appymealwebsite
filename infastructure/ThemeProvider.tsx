import React, { ReactNode } from "react";
import { ThemeProvider as StyledThemeProvider } from "styled-components";
import { theme } from "./theme/index";

interface ThemeProviderProps {
  children: ReactNode;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  return <StyledThemeProvider theme={theme}>{children}</StyledThemeProvider>;
};
