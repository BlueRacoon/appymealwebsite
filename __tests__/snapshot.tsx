import { render } from '@testing-library/react'
import Home from '@/pages/index'
import { ThemeProvider } from '@/infastructure/ThemeProvider'

it('renders homepage unchanged', () => {
  const mockProductDate = {
    _id: 123,
    title: 'abc',
    description: 'abc',
    oldPrice: 123,
    price: 123,
    brand: 'abc',
    image: 'abc',
    isNew: true,
    category: 'abc',
  };
  const mockRestaurantData = {
    id: 'abc',
    name: 'abc',
    username: 'abc',
    city: 'abc',
    zip: 'abc',
    state: 'abc',
    desc: 'abc',
    email: 'abc',
    phoneNumber: 'abc',
    address: 'abc',
    hours: '9-5',
    lat: 123,
    lng: 123,
    accountType: 'abc',
    menus: 'foo',
    isOpen: true,
    isShowing: true,
    createdAt: 'abc',
    photo: 'abc',
    location: {},
    viewport: {},
    fanDiscount: 123,
    fanCount: 123,
    images: [],
    fans: [],
    enableFans: true,
    enableDelivery: true,
    enablePrinting: true,
    deliveryType: {},
    expectedWaitTime: 123,
    menuSelected: 'abc',
    menuHours: 'abc',
    menuStatus: 'foo'
  };
  const { container } = render(
    <ThemeProvider>
      <Home productData={mockProductDate} restaurantData={mockRestaurantData} />
    </ThemeProvider>
  )
  expect(container).toMatchSnapshot()
})
