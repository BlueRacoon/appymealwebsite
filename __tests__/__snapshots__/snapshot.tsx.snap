// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="bg-lightBlue"
  >
    <div
      class="max-w-container mx-auto bg-white"
    >
      <div
        class="w-full h-full bg-[#F1EDEE]  font-titleFont flex border-b-[1px]"
      >
        <div
          class="w-1/2 h-[750px] flex"
        >
          <div
            class="flex flex-col justify-center items-center w-full h-full gap-2"
          >
            <p
              class="font-semibold text-5xl"
            >
              Order Online Now
            </p>
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <a
              href="/restaurants"
            >
              <button
                class="w-72 h-9 bg-white border-[0px] border-dark text-dark   rounded-full flex items-center justify-center gap-1 hover:bg-dark hover:text-white duration-300 shadow-md"
              >
                Order Online
              </button>
            </a>
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <p
              class="font-semibold text-4xl text-dark"
            >
              Discounts & Rewards On App
            </p>
            <div
              class="sc-bgqQcB gNeyFB"
            />
            <a
              href="https://apps.apple.com/us/app/appymeal/id6443683011"
            >
              <button
                class="w-72 h-9 bg-white border-[0px] border-dark text-dark   rounded-full flex items-center justify-center gap-1 hover:bg-dark hover:text-white duration-300 shadow-md"
              >
                Download iOS App
              </button>
            </a>
            <a
              href="https://play.google.com/store/apps/details?id=com.zowen1.AppyMeal&hl=en_US&gl=US"
            >
              <button
                class="w-72 h-9 bg-white border-[0px] border-dark text-dark   rounded-full flex items-center justify-center gap-1 hover:bg-dark hover:text-white duration-300 shadow-md"
              >
                Download Android App
              </button>
            </a>
          </div>
        </div>
        <div
          class="w-1/2 flex relative"
        >
          <img
            alt="LandingPageImg"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
